### PR TITLE
Refactoring of adventure map interface

### PR DESCRIPTION
**Pull request by IvanSavenko**
Initial version of refactoring of adventure map window code.

Goals:
- encapsulation & removal of strong interdependency with CPlayerInterface
- preparation of code for future scalable adventure map window support

### CODE REVIEW

**Comment by krs0** [comment]
LOCPLINT... Could that be renamed A little?

**Comment by IvanSavenko** [comment]
> LOCPLINT... Could that be renamed A little?

Not this time probably...
Not against it, but can't think of any short and understandable name.
Proper name would be "current local player interface", but too long for something that used in half of the code.

Perhaps at least add comment explaining what this thing is so IDE can generate useful tooltip?

File: client/adventureMap/CAdventureMapInterface.cpp [hunk 1]
```diff
@@ -601,89 +531,86 @@ void CAdvMapInt::show(SDL_Surface * to)
...
+	uint32_t scrollDistance = scrollSpeedPixels * timePassed / 1000;
+
+	bool scrollingActive = !GH.isKeyboardCtrlDown() && isActive() && state == EGameState::MAKING_TURN;
+
+	Point cursorPosition = GH.getCursorPosition();
+	Point scrollDirection;
 
-	if(scrollingDir & LEFT)
-		terrain->onMapScrolled(Point(-scrollDistance, 0));
+	if (cursorPosition.x < 15)
```
Comment (Nordsoft91) [inline client/adventureMap/CAdventureMapInterface.cpp:545]: let's move 15 to some const value

**Comment by dydzio0614** [comment]
Will this ease fixing some long-standing hotseat bugs like https://bugs.vcmi.eu/view.php?id=3069 ?

File: client/adventureMap/CAdventureMapInterface.cpp [hunk 2]
```diff
@@ -409,72 +377,56 @@ void CAdvMapInt::fendTurn()
...
-		startIndex = 0;
-	int i = startIndex;
-	do
-	{
-		i++;
-		if (i >= LOCPLINT->wanderingHeroes.size())
-			i = 0;
-	}
-	while (((LOCPLINT->wanderingHeroes[i]->movement == 0) || isHeroSleeping(LOCPLINT->wanderingHeroes[i])) && (i != startIndex));
+	townList->update(town);
```
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply client/adventureMap/CAdventureMapInterface.cpp:418]: townList treats nullptr as "recreate all list".
This behavior had existed before and was not changed in this PR.

File: client/adventureMap/CMinimap.cpp [hunk 3]
```diff
@@ -11,14 +11,15 @@
...
-#include "CAdvMapInt.h"
+#include "CAdventureMapInterface.h"
 
 #include "../widgets/Images.h"
 #include "../CGameInfo.h"
 #include "../CPlayerInterface.h"
 #include "../gui/CGuiHandler.h"
 #include "../render/Colors.h"
 #include "../renderSDL/SDL_PixelAccess.h"
+#include "../render/Canvas.h"
```
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply client/adventureMap/CMinimap.cpp:22]: It is used, and was used before - moved include from header to source file

File: client/adventureMap/CAdventureMapInterface.cpp (hunk 1 above)
Comment (IvanSavenko) [reply client/adventureMap/CAdventureMapInterface.cpp:545]: Will do

**Comment by IvanSavenko** [comment]
> Will this ease fixing some long-standing hotseat bugs like https://bugs.vcmi.eu/view.php?id=3069 ?

Well, at the very least I now know how player interface / adventure map work so can look into stuff like this.
Any other similar issues to this one?

**Comment by dydzio0614** [comment]
Maybe in past I could recall some more, now I cannot

**Comment by dydzio0614** [comment]
there is/was also issue with hotseat battles between 2 players where popup called when clicking spell without eligible target wouldn't appear for one of players

File: test/googletest
[the remark this answers is not in the document]
Comment (IvanSavenko) [reply test/googletest:1]: Done

**Pull request merged**